pyobvector/client/ob_vec_json_table_client.py

```diff
@@ -0,0 +1,776 @@
+        self.perform_raw_text_sql(f"TRUNCATE TABLE {JSON_TABLE_META_TABLE_NAME}")
+        self.jmetadata = ObVecJsonTableClient.JsonTableMetadata(self.user_id)
+    
+    def refresh_metadata(self):
+        self.jmetadata.reflect(self.engine)
+
+    def perform_json_table_sql(self, sql: str):
+        """Perform common SQL that operates on JSON Table."""
+        ast = parse_one(sql, dialect="oceanbase")
+        if isinstance(ast, exp.Create):
+            if ast.kind and ast.kind == 'TABLE':
+                self._handle_create_json_table(ast)
+            else:
+                raise ValueError(f"Create {ast.kind} is not supported")
+        elif isinstance(ast, exp.Alter):
+            self._handle_alter_json_table(ast)
+        elif isinstance(ast, exp.Insert):
+            self._handle_jtable_dml_insert(ast)
+        elif isinstance(ast, exp.Update):
+            self._handle_jtable_dml_update(ast)
+        elif isinstance(ast, exp.Delete):
+            self._handle_jtable_dml_delete(ast)
+        elif isinstance(ast, exp.Select):
+            self._handle_jtable_dml_select(ast)
+        
+    def _parse_datatype_to_str(self, datatype):
+        if datatype == exp.DataType.Type.INT:
+            return "INT"
+        if datatype == exp.DataType.Type.TINYINT:
+            return "TINYINT"
+        if datatype == exp.DataType.Type.TIMESTAMP:
+            return "TIMESTAMP"
+        if datatype == exp.DataType.Type.VARCHAR:
+            return "VARCHAR"
+        if datatype == exp.DataType.Type.DECIMAL:
+            return "DECIMAL"
+        raise ValueError(f"{datatype} not supported")
+    
+    def _calc_default_value(self, default_val):
+        if default_val is None:
+            return None
+        with self.engine.connect() as conn:
+            res = conn.execute(text(f"SELECT {default_val}"))
+            for r in res:
+                logger.info(f"============== Calculate default value: {r[0]}")
+                return r[0]
+    
+    def _handle_create_json_table(self, ast: Expression):
+        logger.info("HANDLE CREATE JSON TABLE")
+
+        if not isinstance(ast.this, exp.Schema):
+            raise ValueError("Invalid create table statement")
+        schema = ast.this
+        if not isinstance(schema.this, exp.Table):
+            raise ValueError("Invalid create table statement")
+        jtable = schema.this
+        if not isinstance(jtable.this, exp.Identifier):
+            raise ValueError("Invalid create table statement")
+        jtable_name = jtable.this.this
+        logger.info(jtable_name)
+
+        if jtable_name == JSON_TABLE_META_TABLE_NAME or jtable_name == JSON_TABLE_DATA_TABLE_NAME:
+            raise ValueError(f"Invalid table name: {jtable_name}")
+        if jtable_name in self.jmetadata.meta_cache:
+            raise ValueError("Table name duplicated")
+        
+        Session = sessionmaker(bind=self.engine)
+        session = Session()
+        new_meta_cache_items = []
+        col_id = 16
```

Comment:
why does `col_id` starts with 16?

Reply:
The `OceanBase` user column ID also starts with 16, which can be considered an easter egg.